src/toil/jobStores/azureJobStore.py

```diff
@@ -135,7 +148,13 @@ def create(self, command, memory, cores, disk,
                        predecessorNumber=predecessorNumber)
         entity = job.toItem(chunkSize=self.jobChunkSize)
         entity['RowKey'] = jobStoreID
-        self.jobItems.insert_entity(entity=entity)
+        try:
+            self.jobItems.insert_entity(entity=entity)
+        except WindowsAzureConflictError as e:
+            # A conflict on insert probably means we retried but already
+            # succeeded. Just ignore it. We made a _newJobID() ourselves, so
+            # unless that's broken we can't be re-using a job ID.
+            logger.warning("Conflict when inserting job %s" % jobStoreId)
```

Comment:
Although useful, this is unrelated to the issue that this PR is solving. It seems to partially address #579, partially because this here isn't the only call of insert_entity.